source/include/core_mqtt.h

```diff
@@ -101,6 +107,61 @@ typedef void (* MQTTEventCallback_t )( struct MQTTContext * pContext,
                                        struct MQTTPacketInfo * pPacketInfo,
                                        struct MQTTDeserializedInfo * pDeserializedInfo );
 
+/**
+ * @brief User defined callback used to store outgoing publishes. Used to track any publish
+ * retransmit on an unclean session connection.
+ *
+ * @param[in] pContext Initialised MQTT Context.
+ * @param[in] packetId Outgoing publish packet identifier.
+ * @param[in] pMqttVec Pointer to the opaque mqtt vector structure. Users should use MQTT_SerializeMQTTVec
+ *                and MQTT_GetBytesInMQTTVec functions to get the memory required and to serialize the
+ *                MQTTVec_t in the provided memory respectively.
+ * @param[in] mqttVecCount Number of transport vectors in the pIoVec array.
+ *
+ * @return True if the copy is successful else false.
+ */
+/* @[define_mqtt_retransmitstorepacket] */
+typedef bool ( * MQTTStorePacketForRetransmit)( struct MQTTContext * pContext,
+                                                uint16_t packetId,
+                                                MQTTVec_t * pMqttVec,
+                                                size_t mqttVecCount );
```

Comment:
shouldn't mqttVecCount be part of the MQTTVec_t? Seems like leaking an internal detail

Reply:
Fixed